src/ai/backend/common/events.py

```diff
@@ -194,7 +195,7 @@ def deserialize(cls, value: tuple):
 class KernelCreationEventArgs:
     kernel_id: KernelId = attr.ib()
     creation_id: str = attr.ib()
-    reason: str = attr.ib(default="")
+    reason: KernelLifecycleEventReason | str = attr.ib(default="")
```

Comment:
It'll be better to just set `KernelLifecycleEventReason` as the only allowed type of `reason`, add `UNKNOWN` value to the Enum, and set that value as default of `reason` since it seems better not to allow any arbitrary reasons.
```suggestion
    reason: KernelLifecycleEventReason = attr.ib(default=KernelLifecycleEventReason.UNKNOWN)
```